pkg/cloud/awsprovider.go

```diff
@@ -2314,3 +2315,15 @@ func (aws *AWS) PricingSourceSummary() interface{} {
 	// encode the pricing source summary as a JSON string
 	return aws.Pricing
 }
+
+func (aws *AWS) GetNodePoolName(labels map[string]string) string {
```

Comment:
Could this be function be moved into provider.go so it's not repeated across providers?

Reply:
An interesting proposal! I guess the reason I went with this implementation is that the label we use to determine the name of the node pool is different for each provider. For example, the label in azure is kubernetes.azure.com/agentpool, whereas in AWS it is  eks.amazonaws.com/nodegroup. Let me look at the approach again and see if I can optimize

Reply:
That makes sense! I'm interested to see what you come up with.

Reply:
@nik-kc I took a stab at a new implementation. Interface func now just returns the bare minimum provider-specific label, all common logic refactored to the generic provider - see what you think!

Reply:
Chef's kiss, love it. Great work!